### PR TITLE
knex - Added missing schema builder parameters

### PR DESCRIPTION
**Pull request by rmblstrp**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not provide its own types, and you can not add them.
- [ ] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: http://knexjs.org/#Schema-primary http://knexjs.org/#Schema-foreign http://knexjs.org/#Schema-unique http://knexjs.org/#Schema-timestamp
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

If removing a declaration:
- [ ] If a package was never on DefinitelyTyped, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

File: types/knex/index.d.ts [hunk 1]
```diff
@@ -13,6 +13,7 @@
 import events = require("events");
 import stream = require ("stream");
 import Bluebird = require("bluebird");
+import { ColumnBuilder } from "knex";
```
Comment (mastermatt) [inline types/knex/index.d.ts]: Your IDE may have auto added this on accident.

File: types/knex/index.d.ts [hunk 2]
```diff
@@ -439,8 +440,8 @@ declare namespace Knex {
         primary(columnNames: string[]): TableBuilder;
         index(columnNames: (string | Raw)[], indexName?: string, indexType?: string): TableBuilder;
         unique(columnNames: (string | Raw)[], indexName?: string): TableBuilder;
-        foreign(column: string): ForeignConstraintBuilder;
-        foreign(columns: string[]): MultikeyForeignConstraintBuilder;
+        foreign(column: string, foreignKeyName?: string): ForeignConstraintBuilder;
+        foreign(columns: string[], foreignKeyName?: string): ForeignConstraintBuilder;
```
Comment (mastermatt) [inline types/knex/index.d.ts]: Was the `MultikeyForeignConstraintBuilder` a bug or is the return type a `ForeignConstraintBuilder` when `foreignKeyName` is provided and a `MultikeyForeignConstraintBuilder` when omitted?

**Review by mastermatt — CHANGES_REQUESTED** [review]
Could you add some tests showing these usages please.

File: types/knex/index.d.ts (hunk 1 above)
Comment (rmblstrp) [reply types/knex/index.d.ts]: Very likely.  I'll remove that.

File: types/knex/index.d.ts (hunk 2 above)
Comment (rmblstrp) [reply types/knex/index.d.ts]: That would be a mistake on my part.  I will correct.

**Comment by rmblstrp** [comment]
I will fix the noted issues and add usage tests.

**Review by mastermatt — APPROVED** [review]

**Comment by DanielRosenwasser** [comment]
Thanks!

**Pull request merged**